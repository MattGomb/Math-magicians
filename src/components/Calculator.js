import React from 'react';
import calculate from '../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      operation: null,
      next: null,
      error: null,
    };
    this.click = this.click.bind(this);
  }

  click = (e) => {
    const result = calculate(this.state, e.target.textContent);
    this.setState(result);
  }

  render() {
    const {
      total, operation, next, error,
    } = this.state;
    return (
      <div className="calculator-container">
        <div className="result">
          {total}
          {operation}
          {next}
          {error}
        </div>
        <button type="button" onClick={this.click}>AC</button>
        <button type="button" onClick={this.click}>+/-</button>
        <button type="button" onClick={this.click}>%</button>
        <button type="button" className="last-col" onClick={this.click}>รท</button>
        <button type="button" onClick={this.click}>7</button>
        <button type="button" onClick={this.click}>8</button>
        <button type="button" onClick={this.click}>9</button>
        <button type="button" className="last-col" onClick={this.click}>*</button>
        <button type="button" onClick={this.click}>4</button>
        <button type="button" onClick={this.click}>5</button>
        <button type="button" onClick={this.click}>6</button>
        <button type="button" className="last-col" onClick={this.click}>-</button>
        <button type="button" onClick={this.click}>1</button>
        <button type="button" onClick={this.click}>2</button>
        <button type="button" onClick={this.click}>3</button>
        <button type="button" className="last-col" onClick={this.click}>+</button>
        <button type="button" className="zero" onClick={this.click}>0</button>
        <button type="button" onClick={this.click}>.</button>
        <button type="button" className="last-col" onClick={this.click}>=</button>
      </div>
    );
  }
}

export default Calculator;
