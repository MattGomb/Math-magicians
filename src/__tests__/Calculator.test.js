import Calculator from '../components/Calculator';
import React from 'react';
import renderer from 'react-test-renderer';

describe('the calculator snapshot and react testing library', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});