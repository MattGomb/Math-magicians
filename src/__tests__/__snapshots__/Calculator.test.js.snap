// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the calculator snapshot and react testing library renders correctly 1`] = `
<section
  className="calculator-container"
>
  <div
    className="title"
  >
    Just MATH it ✔
  </div>
  <div
    className="calculator-app-container"
  >
    <div
      className="result"
    >
      0
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="last-col"
      onClick={[Function]}
      type="button"
    >
      ÷
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="last-col"
      onClick={[Function]}
      type="button"
    >
      *
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="last-col"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="last-col"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="last-col"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</section>
`;
